### PR TITLE
feat(parser): add ability to get data to CSV or JSON

### PR DESCRIPTION
**Pull request by PierreReliquet**

### CODE REVIEW

**Review by PierreReliquet — COMMENTED** [review]
@antoine-richard don't know if you can review the code?

File: parser/index.js [hunk 1]
```diff
@@ -12,7 +12,17 @@ if (args.length !== 2) {
...
 }
 
-mkdir('out')
+mkdirIfRequired('out')
     .then(() => readFile(args[0], 'utf8'))
-    .then(data => parse(data))
+    .then(data => parser.parse(data))
     .then(csvContent => writeFile('out/' + args[1], csvContent, 'utf8'));
+
+function mkdirIfRequired(path) {
```
Comment (antoine-richard) [inline parser/index.js:20]: yes saw this issue too.
an alternative is to use mkdirp module (which does that for you)

File: parser/parse.js [hunk 2]
```diff
@@ -1,27 +1,59 @@
...
+ * Formats the data with the provided format if no
+ * formatter is defined then it falls back to the CSV one.
+ */
+function parse(data, formatter) {
+    if (!formatter) {
+        formatter = FORMATTERS.CSV;
     }
-    return Promise.resolve(result);
+
+    var jsonKills = _(data.match(KILL_REGEX_GLOBAL))
```
Comment (antoine-richard) [inline parser/parse.js:24]: nice refactoring.
I'd rename jsonKills to killsData because it could either be json or csv.

Comment (antoine-richard) [reply parser/parse.js:24]: ok no actually it's json.
so i had issues to understand that we parse in json by default, and convert it to csv if required.
I wonder how we could improve readability (remove the surprise).

**Review by antoine-richard — COMMENTED** [review]
posted comments with suggestions, but they're not mandatory to merge.
we can merge and improve later. (first do it, then do it right, then do it better)

**Review by antoine-richard — APPROVED** [review]

File: parser/parse.js (hunk 2 above)
Comment (PierreReliquet) [reply parser/parse.js:24]: I went for the following behaviour for now:
1. generate a standard json array
2. transform it to any expected output which for now is very limited 😛

File: parser/index.js (hunk 1 above)
Comment (PierreReliquet) [reply parser/index.js:20]: did know that 😉 thanks for the tip, I'll keep this one for now

**Pull request merged**